src/components/CreateNewProjectWizard/Gatsby/ProjectStarterSelection.js

```diff
@@ -0,0 +1,78 @@
+// @flow
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+
+import TextInputWithButton from '../../TextInputWithButton';
+
+import * as actions from '../../../actions';
+import type { Dispatch } from '../../../actions/types';
+
+type State = {
+  gatsbyStarter: string,
+};
+
+type Props = {
+  projectStarter: string,
+  isFocused: boolean,
+  handleFocus: string => void,
+  onSelect: string => void,
+  onFocus: () => void,
+  showStarterSelection: Dispatch<typeof actions.showStarterSelectionModal>,
+};
+
+class ProjectStarter extends PureComponent<Props, State> {
+  state = {
+    gatsbyStarter: '',
+  };
+
+  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
+    return {
+      gatsbyStarter: nextProps.projectStarter,

```

Comment:
I think you're right but with-out this it would require me to modify modal state as I need to have the info about the entered starter before opening the modal.
So the user can dismiss the changes made in the modal. Anyway I'm not changing this as I'll check if I can remove the second modal - so this won't be needed anymore.

Reply:
It's acutally an anti-pattern to override local state unconditionally with every prop change.
There is a nice post [you-probably-dont-need-derived-state](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html) about this in the React.js blog.

We don't need it here anymore as I'm rewriting this. But I wanted to note this as I've learned this and I wasn't aware of it.